client/src/components/admin-dashboard-component/index.jsx

```diff
@@ -0,0 +1,278 @@
+import * as React from "react";
+import { useNavigate } from "react-router-dom";
+import PropTypes from "prop-types";
+import Box from "@mui/material/Box";
+import { DataGrid, GridActionsCellItem, gridClasses } from "@mui/x-data-grid";
+import Button from "@mui/material/Button";
+import AddIcon from "@mui/icons-material/Add";
+import EditIcon from "@mui/icons-material/Edit";
+import DeleteIcon from "@mui/icons-material/DeleteOutlined";
+import Card from "@mui/material/Card";
+import CardActions from "@mui/material/CardActions";
+import CardContent from "@mui/material/CardContent";
+import TextField from "@mui/material/TextField";
+import InputAdornment from "@mui/material/InputAdornment";
+import SearchIcon from "@mui/icons-material/Search";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import Typography from "@mui/material/Typography";
+import UserType from "../../prop-types/UserType";
+import ConfirmDialog from "../shared/confirm-dialog-component";
+import LoadingComponent from "../shared/loading-screen-component";
+
+import { DashboardContainer, HeaderContainer } from "./index.styles";
+import { deleteUser } from "../../utils/api";
+import ErrorComponent from "../shared/error-screen-component";
+
+function AdminDashboardComponent({ managedUsers, removeUser }) {
+  const navigate = useNavigate();
+  const [searchQuery, setSearchQuery] = React.useState("");
+  const [filteredRows, setFilteredRows] = React.useState(managedUsers);
+  const [confirmDeleteDialog, setConfirmDeleteDialog] = React.useState(false);
+  const [userToDelete, setUserToDelete] = React.useState(null);
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState("");
+
+  React.useEffect(() => {
+    setFilteredRows(managedUsers);
+  }, [managedUsers]);
+
+  const handleSearch = (event) => {
+    const query = event.target.value;
+    setSearchQuery(query);
+
+    const filtered = managedUsers.filter((row) => {
+      return row.displayName.toLowerCase().includes(query.toLowerCase());
+    });
+    setFilteredRows(filtered);
+  };
+
+  const handleFilterReset = () => {
+    setSearchQuery("");
+    setFilteredRows(managedUsers);
+  };
+
+  const handleBackClick = () => {
+    navigate("/dashboard");
+  };
+
+  const handleDeleteClick = (userID) => {
+    setUserToDelete(userID);
+    setConfirmDeleteDialog(true);
+  };
+
+  const handleConfirmDelete = async () => {
+    if (userToDelete !== null) {
+      setIsLoading(true);
```

Comment:
Remove anything relating to the loading state in this component. This makes it so that whenever a user is deleted a loading page pops up. Removing this makes the deletion flow much smoother.

Reply:
I agree thanks.